backend/degree/serializers.py

```diff
@@ -89,14 +99,23 @@ class Meta:
 
 class FulfillmentSerializer(serializers.ModelSerializer):
     course = serializers.SerializerMethodField()
+
     def get_course(self, obj):
-        course = Course.with_reviews.filter(full_code=obj.full_code, semester__lte=get_current_semester()).order_by("-semester").first()
+        course = (
+            Course.with_reviews.filter(
+                full_code=obj.full_code, semester__lte=get_current_semester()
+            )
+            .order_by("-semester")
+            .first()
+        )
```

Comment:
technically `Min` might will generate a better query for the db but not a big deal